test/runtests.jl

```diff
@@ -124,13 +124,13 @@ if VERSION >= v"0.5.0-dev+3305"
 	@test Base.iteratorsize(ch3) == Base.HasLength()
 end
 
-r = ( x for x in 1:10 if isodd(x))
-ch4 = chain(1:10, r)
 if VERSION >= v"0.5.0-dev+3305"
+	r = ( x for x in 1:10 if isodd(x))
```

Comment:
Even though this wouldn't actually run on v0.4, the compiler still attempts to parse it, and unfortunately, this syntax is a parse error on v0.4. 😢 

For now, my suggestion is to comment out this test, explaining that it is a parse error on Julia v0.4, and that it should be uncommented when support for v0.4 is dropped.